browser/resources/federated_internals/federated_internals.js

```diff
@@ -0,0 +1,126 @@
+import {$} from 'chrome://resources/js/util.m.js';
+import { FederatedInternalsBrowserProxy } from './federated_internals_browser_proxy.js';
+
+const dataStoresLogs = {};
+let selectedDataStore = 'ad-timing';
+
+function getProxy() {
+  return FederatedInternalsBrowserProxy.getInstance();
+}
+
+function initialize() {
+  getProxy().getAdStoreInfo();
+  
+  getProxy().getCallbackRouter().onAdStoreInfoAvailable.addListener(
+    (logs) => {
+      dataStoresLogs['ad-timing'] = logs;
+      onDataStoreChanged();
+  });
+
+  const button = $('data-store-logs-dump');
+  button.addEventListener('click', onLogsDump);
+
+  $('stores').onchange = onDataStoreChanged;
+}
+
+function onDataStoreChanged() {
+    let store_selection = $('stores')
+    selectedDataStore = store_selection.options[store_selection.selectedIndex].value;
+    const logs = dataStoresLogs[selectedDataStore];
+
+    if (logs.length == 0) {
+      $('service-message').textContent = "No logs for selected Data Store."
+      return;
+    }
+
+    if (selectedDataStore == 'ad-timing') {
+      $('service-message').textContent = ''
+      Object.keys(logs[0]).forEach(function(title) {
+          const th = document.createElement('th');
+          th.textContent = title;
+          th.className = 'ad-timing-log-'+ title;
+      
+          const thead = $('ad-timing-headers');
+          thead.appendChild(th);
+      });
+    
+      logs?.forEach(function(log) {
```

Comment:
why do we need an optional chaining here? logs object is already accessed earlier